packages/ckeditor5-link/src/ui/linkformview.ts

```diff
@@ -242,49 +282,146 @@ export default class LinkFormView extends View {
 	}
 
 	/**
-	 * Creates a labeled input view.
-	 *
-	 * @returns Labeled field view instance.
+	 * Creates a back button view that cancels the form.
 	 */
-	private _createUrlInput(): LabeledFieldView<InputTextView> {
+	private _createBackButton(): ButtonView {
 		const t = this.locale!.t;
-		const labeledInput = new LabeledFieldView( this.locale, createLabeledInputText );
+		const backButton = new ButtonView( this.locale );
 
-		labeledInput.fieldView.inputMode = 'url';
-		labeledInput.label = t( 'Link URL' );
+		backButton.set( {
+			label: t( 'Cancel' ),
+			icon: icons.previousArrow,
+			tooltip: true
+		} );
 
-		return labeledInput;
+		backButton.delegate( 'execute' ).to( this, 'cancel' );
+
+		return backButton;
 	}
 
 	/**
-	 * Creates a button view.
-	 *
-	 * @param label The button label.
-	 * @param icon The button icon.
-	 * @param className The additional button CSS class name.
-	 * @param eventName An event name that the `ButtonView#execute` event will be delegated to.
-	 * @returns The button view instance.
-	 */
-	private _createButton( label: string, icon: string, className: string, eventName?: string ): ButtonView {
-		const button = new ButtonView( this.locale );
-
-		button.set( {
-			label,
-			icon,
+	 * Creates a settings button view that opens the advanced settings panel.
+	 */
+	private _createSettingsButton(): ButtonView {
+		const t = this.locale!.t;
+		const settingsButton = new ButtonView( this.locale );
+
+		settingsButton.set( {
+			label: t( 'Advanced' ),
+			icon: icons.settings,
 			tooltip: true
 		} );
 
-		button.extendTemplate( {
+		return settingsButton;
+	}
+
+	/**
+	 * Creates a save button view that inserts the link.
+	 */
+	private _createSaveButton(): ButtonView {
+		const t = this.locale!.t;
+		const saveButton = new ButtonView( this.locale );
+
+		saveButton.set( {
+			label: t( 'Insert' ),
```

Comment:
To remember - probably we will need to change the label when link already exists (editing). @dwandeljanicka am i right?